### PR TITLE
[SPARK-37120][BUILD] Add Daily GitHub Action jobs for Java11/17

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
### What changes were proposed in this pull request?

This PR aims to add Daily GitHub Action jobs for Java 11/17.

### Why are the changes needed?

To add a test coverage on Java 11/17.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

N/A.

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
cc @HyukjinKwon

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -33,12 +33,17 @@ on:
     - cron: '0 7 * * *'
     # PySpark coverage for master branch
     - cron: '0 10 * * *'
+    # Java 11
+    - cron: '0 13 * * *'
+    # Java 17
+    - cron: '0 16 * * *'
```
Comment (dongjoon-hyun) [inline .github/workflows/build_and_test.yml:39]: Unfortunately, we cannot test the daily jobs via PR Builder or [personal fork](https://github.com/dongjoon-hyun/spark/actions/workflows/build_and_test.yml?query=event%3Aschedule).

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Jobs are running properly. Let me get this in and see how it gose - seems like PR builders don't get affected anyway.

**Comment by HyukjinKwon** [comment]
Merged to master.

**Comment by dongjoon-hyun** [comment]
Thank you so much, @HyukjinKwon !

**Comment by LuciferYang** [comment]
Thanks for ping me @dongjoon-hyun. LGTM +1

**Pull request closed without merging**